### PR TITLE
Add support for Google searches using Google.Apis.Customsearch.v1

### PR DESCRIPTION
**Pull request by ScottKane**
### Motivation and Context
Allow the usage of Google search equivalent to current Bing search implementation.

### Description
Uses the Google.Apis.Customsearch.v1 packages to interface with the Google search API, this requires an API key from GCP and a custom search engine ID

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by ScottKane** [comment]
@microsoft-github-policy-service agree

**Comment by ScottKane** [comment]
> @ScottKane please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by ScottKane** [comment]
@microsoft-github-policy-service agree

File: dotnet/src/SemanticKernel.Skills/Skills.Web/Google/GoogleConnector.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace Microsoft.SemanticKernel.Skills.Web.Google;
+
+public class GoogleConnector : IWebSearchEngineConnector, IDisposable
+{
+    private readonly ILogger             _logger;
+    private readonly CustomsearchService _search;
+    private readonly string              _searchEngineId;
+
+    public GoogleConnector(string apiKey, string searchEngineId, ILogger<GoogleConnector>? logger = null)
```
Comment (awharrison-28) [inline dotnet/src/SemanticKernel.Skills/Skills.Web/Google/GoogleConnector.cs:20]: Do you mind calling this GoogleSearchConnector?

Comment (ScottKane) [reply dotnet/src/SemanticKernel.Skills/Skills.Web/Google/GoogleConnector.cs:20]: Not at all, I was just going off the BingConnector. If that's the case, should that be renamed to BingSearchConnector too?

Comment (ScottKane) [reply dotnet/src/SemanticKernel.Skills/Skills.Web/Google/GoogleConnector.cs:20]: I would probably also suggest a change to `IWebSearchEngineConnector` to allow for searches returning multiple results, currently both implementations return the first result. I would suggest an additional parameter for the maximum number of results returned defaulting to 1, which should mitigate any breaking change

**Comment by MovGP0** [comment]
A few thoughts:

- the `this.` keywords can be safely removed everywhere, since they do not contribute anything.
- the exact version of the `Google.Apis.Customsearch.v1` package should not be hardcoded, such that bugfixes of the library will not be missed in future builds. Only fix the `MAJOR.MINOR` part of the version.
- consider setting `<ImplicitUsings>true</ImplicitUsings>` in order to clean up the using section

**Comment by ScottKane** [comment]
The original code written was without `this.` however StyleCop was enforcing the use of them so I assumed this was required. Happy to make any changes required as soon as we can iron out the listed questions surrounding the interface and naming conventions. The `BingConnector` also takes a string `apiKey` so should that change to an Option type too?

**Comment by adrianwyatt** [comment]
> The original code written was without `this.` however StyleCop was enforcing the use of them so I assumed this was required. Happy to make any changes required as soon as we can iron out the listed questions surrounding the interface and naming conventions. The `BingConnector` also takes a string `apiKey` so should that change to an Option type too?

Yes, current repo style guides are to use "this.". You should also run "dotnet format" in the "/dotnet" directory to gets most of the style adjustments made for free. This will help the "dotnet-format" gate pass.

**Comment by ScottKane** [comment]
Ok I ran `dotnet format` as requested, I think ommitting `Search` from the Bing/Google connectors is a better descision, they both implement `IWebSearchEngineConnector` which can tell you're working with a search engine while leaving the posibility for the same connectors to implement different interfaces in the future, e.g both could implement an `IWebMapsConnector` or something, at which point having `Search` in the name might not be so sensible.

**Comment by dluc** [comment]
@ScottKane thanks for bringing this in! Unfortunately you unchecked the "allow maintainers to edit the PR" and we could not address some important changes, e.g. the dependency on an old nuget. I tried working around the block but between this and several conflicts it was taking too long, so I recreated your PR here https://github.com/microsoft/semantic-kernel/pull/737 and added you as one of the contributors in the commit log.

**Comment by ScottKane** [comment]
@dluc no problem, sorry I will look out for that next time I raise a PR

**Pull request merged**